backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/OperationInterpreterViewSwitch.java

```diff
@@ -0,0 +1,206 @@
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.emf.view;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.sirius.components.core.api.IEditService;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.emf.services.EditingContext;
+import org.eclipse.sirius.components.interpreter.AQLInterpreter;
+import org.eclipse.sirius.components.interpreter.Result;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.components.view.ChangeContext;
+import org.eclipse.sirius.components.view.CreateInstance;
+import org.eclipse.sirius.components.view.DeleteElement;
+import org.eclipse.sirius.components.view.Operation;
+import org.eclipse.sirius.components.view.SetValue;
+import org.eclipse.sirius.components.view.UnsetValue;
+import org.eclipse.sirius.components.view.util.ViewSwitch;
+import org.eclipse.sirius.ecore.extender.business.internal.accessor.ecore.EcoreIntrinsicExtender;
+
+/**
+ * This switch allows to handle common {@link Operation}s without dependencies to a specific representation.
+ *
+ * @author fbarbin
+ */
+public class OperationInterpreterViewSwitch extends ViewSwitch<Optional<VariableManager>> {
+    protected final VariableManager variableManager;
+
+    protected final AQLInterpreter interpreter;
+
+    protected final IEditService editService;
+
+    protected final IOperationInterpreter operationInterpreter;
+
+    protected final EcoreIntrinsicExtender ecore;
+
+    /**
+     * Default constructor.
+     *
+     * @param variableManager
+     *            the current {@link VariableManager}.
+     * @param interpreter
+     *            the {@link AQLInterpreter}.
+     * @param editService
+     *            the {@link IEditService}.
+     * @param operationInterpreter
+     *            the {@link IOperationInterpreter} used for delegating sub operations executions. It is the
+     *            responsibility of the {@link IOperationInterpreter} to delegate each operation execution to the
+     *            appropriate switch according to the concrete representation.
+     */
+    public OperationInterpreterViewSwitch(VariableManager variableManager, AQLInterpreter interpreter, IEditService editService, IOperationInterpreter operationInterpreter) {
+        this.variableManager = variableManager;
```

Comment:
Is that normal that this can be `null`? Because the code of the methods do not seem to consider `null` as a valid use case

Reply:
no it is not. I'll add a requireNonNull check